qiskit/opflow/gradients/hessian.py

```diff
@@ -214,14 +213,14 @@ def is_coeff_c(coeff, c):
             # and params = (θ0,θ1)
             dd_ops = [self.get_hessian(op, params) for op in operator.oplist]
 
-            # Note that this check to see if the ListOp has a default combo_fn
+            # TODO Note that this check to see if the ListOp has a default combo_fn
             # will fail if the user manually specifies the default combo_fn.
             # I.e operator = ListOp([...], combo_fn=lambda x:x) will not pass this check and
             # later on jax will try to differentiate it and fail.
             # An alternative is to check the byte code of the operator's combo_fn against the
             # default one.
             # This will work but look very ugly and may have other downsides I'm not aware of
-            if operator.combo_fn == ListOp([]).combo_fn:
+            if operator.combo_fn == ListOp([]).combo_fn:  # pylint: disable=comparison-with-callable
```

Comment:
Do we know why this wasn't triggering an error before?  Is it necessary?

Reply:
I tested with a updated pylint to make sure that the lines I removed in this PR wouldn't just need adding back whenever we upgrade the pylint pin. I guess this is one that gets warned by updated pylint and I accidentally added the supression to this PR. I'll drop it for now.